### ISSUE TITLE
mcedit2 v gd4f1656  (Linux x64, Ubuntu 15.04) - Crash On Load

### ISSUE THREAD

**Issue by EvilSupahFly** [mcedit/mcedit2#169]
Contents of `/home/seann/mcedit_v2/MCEdit User Data/mcedit.log`:
```
[INFO][main.py:165]:MCEdit2 version v2.0-tech-demo-994-gd4f1656 starting...
[INFO][acceleratesupport.py:12]:OpenGL_accelerate module loaded
[WARNING][arraydatatype.py:17]:Unable to load ArrayDatatype accelerator from OpenGL_accelerate
[WARNING][converters.py:122]:Unable to load converters accelerators (wrapper, arraydatatype) from OpenGL_accelerate
[WARNING][arrayhelpers.py:22]:Unable to load arrayhelpers accelerator from OpenGL_accelerate
[WARNING][numpymodule.py:28]:Unable to load numpy_formathandler accelerator from OpenGL_accelerate
[INFO][formathandler.py:54]:Unable to load registered array format handler numeric:
Traceback (most recent call last):
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/arrays/formathandler.py", line 44, in loadPlugin  
    plugin_class = entrypoint.load()
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/plugins.py", line 14, in load (self is a FormatHandler)
    return importByName( self.import_path )
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/plugins.py", line 28, in importByName  
    module = __import__( ".".join(moduleName), {}, {}, moduleName)
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/arrays/numeric.py", line 15, in <module>  
    raise ImportError( """No Numeric module present: %s"""%(err))
ImportError: No Numeric module present: No module named Numeric

[INFO][settings.py:100]:Loading app settings from /home/seann/mcedit_v2/MCEdit User Data/mcedit2.ini
[INFO][loadablechunks.py:17]:Making checkerboard texture...
[ERROR][main.py:187]:Unhandled Exception: 
	No module named paths
Traceback (most recent call last):
  File "/usr/local/bin/mcedit2", line 9, in <module>  
    load_entry_point('mcedit2==2.0.0a1.dev0', 'console_scripts', 'mcedit2')()
  File "/home/seann/mcedit_v2/src/mcedit2/main.py", line 204, in main  
    app = startup()
  File "/home/seann/mcedit_v2/src/mcedit2/util/profiler.py", line 66, in _wrapper (self is a Profiler)
    return func(*a, **kw)
  File "/home/seann/mcedit_v2/src/mcedit2/main.py", line 176, in startup  
    from mcedit2.editorapp import MCEditApp
  File "/home/seann/mcedit_v2/src/mcedit2/editorapp.py", line 34, in <module>  
    from mcedit2.util.ipython_widget import terminal_widget
  File "/home/seann/mcedit_v2/src/mcedit2/util/ipython_widget.py", line 67, in <module>  
    from qtconsole.inprocess import QtInProcessKernelManager
  File "/home/seann/.local/lib/python2.7/site-packages/qtconsole/inprocess.py", line 5, in <module>  
    from ipykernel.inprocess import (
  File "/home/seann/.local/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>  
    from .connect import *
  File "/home/seann/.local/lib/python2.7/site-packages/ipykernel/connect.py", line 13, in <module>  
    from IPython.paths import get_ipython_dir
ImportError: No module named paths
```
Terminal output (including some investigative probing):
```
seann@black-beast:~/mcedit_v2 $ mcedit2 --debug
Logging level set
Logging to /home/seann/mcedit_v2/MCEdit User Data/mcedit.log
Setting up console handler
[WARNING][arraydatatype.py:17]:Unable to load ArrayDatatype accelerator from OpenGL_accelerate
[WARNING][converters.py:122]:Unable to load converters accelerators (wrapper, arraydatatype) from OpenGL_accelerate
[WARNING][arrayhelpers.py:22]:Unable to load arrayhelpers accelerator from OpenGL_accelerate
[WARNING][numpymodule.py:28]:Unable to load numpy_formathandler accelerator from OpenGL_accelerate
[ERROR][main.py:187]:Unhandled Exception: 
	No module named paths
Traceback (most recent call last):
  File "/usr/local/bin/mcedit2", line 9, in <module>  
    load_entry_point('mcedit2==2.0.0a1.dev0', 'console_scripts', 'mcedit2')()
  File "/home/seann/mcedit_v2/src/mcedit2/main.py", line 204, in main  
    app = startup()
  File "/home/seann/mcedit_v2/src/mcedit2/util/profiler.py", line 66, in _wrapper (self is a Profiler)
    return func(*a, **kw)
  File "/home/seann/mcedit_v2/src/mcedit2/main.py", line 176, in startup  
    from mcedit2.editorapp import MCEditApp
  File "/home/seann/mcedit_v2/src/mcedit2/editorapp.py", line 34, in <module>  
    from mcedit2.util.ipython_widget import terminal_widget
  File "/home/seann/mcedit_v2/src/mcedit2/util/ipython_widget.py", line 67, in <module>  
    from qtconsole.inprocess import QtInProcessKernelManager
  File "/home/seann/.local/lib/python2.7/site-packages/qtconsole/inprocess.py", line 5, in <module>  
    from ipykernel.inprocess import (
  File "/home/seann/.local/lib/python2.7/site-packages/ipykernel/__init__.py", line 2, in <module>  
    from .connect import *
  File "/home/seann/.local/lib/python2.7/site-packages/ipykernel/connect.py", line 13, in <module>  
    from IPython.paths import get_ipython_dir
ImportError: No module named paths
QWidget: Must construct a QApplication before a QPaintDevice
Aborted (core dumped)

seann@black-beast:~/mcedit_v2 $ pip list | grep ipython
ipython (3.0.0)
ipython-genutils (0.1.0)

seann@black-beast:~/mcedit_v2 $ python -m IPython --version
3.2.1

seann@black-beast:~/mcedit_v2 $ cat /usr/local/lib/python2.7/dist-packages/*.pth
import sys; sys.__plen = len(sys.path)
/home/seann/mcedit_v2/src
import sys; new = sys.path[sys.__plen:]; del sys.path[sys.__plen:]; p = getattr(sys, '__egginsert', 0); sys.path[p:p] = new; sys.__egginsert = p + len(new)
```
The only thing that's changed is the version of MCE2 I'm running now (downloaded gd4f1656 this morning).
```
seann@black-beast:~/mcedit_v2 $ uname -a
Linux black-beast 3.19.0-31-generic #36-Ubuntu SMP Wed Oct 7 15:04:02 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

![screenshot from 2015-11-09 12 50 47](https://cloud.githubusercontent.com/assets/3280846/11041560/9287063c-86e0-11e5-8ef4-f3de46f22350.png)

**Comment by codewarrior0** [comment]
Hmm. Try `pip install -U ipython qtconsole`

I bumped my IPython up to 4.0.x a few days ago. The only thing I had to change was a few imports, since 4.0 was split up into several packages.

**Comment by EvilSupahFly** [comment]
Well, it starts to load now, but now it crashes as it generates preview of first item in list upon initial start-up. 

Terminal output:
```
seann@black-beast:~/mcedit_v2 $ mcedit2
Logging level set
Logging to /home/seann/mcedit_v2/MCEdit User Data/mcedit.log
Setting up console handler
[WARNING][arraydatatype.py:17]:Unable to load ArrayDatatype accelerator from OpenGL_accelerate
[WARNING][converters.py:122]:Unable to load converters accelerators (wrapper, arraydatatype) from OpenGL_accelerate
[WARNING][arrayhelpers.py:22]:Unable to load arrayhelpers accelerator from OpenGL_accelerate
[WARNING][numpymodule.py:28]:Unable to load numpy_formathandler accelerator from OpenGL_accelerate
[WARNING][editorapp.py:92]:UserFilesDirectory: /home/seann/mcedit_v2/MCEdit User Data
[WARNING][configureblocksdialog.py:119]:Failed to read definitions file defined_blocks.json
Xlib: request 43 length 4 would exceed buffer size.
[xcb] Unknown request in queue while dequeuing
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
python: ../../src/xcb_io.c:179: dequeue_pending_request: Assertion `!xcb_xlib_unknown_req_in_deq' failed.
Aborted (core dumped)
```
Contents of `/home/seann/mcedit_v2/MCEdit User Data/mcedit.log`:
```
[INFO][main.py:165]:MCEdit2 version v2.0-tech-demo-994-gd4f1656 starting...
[INFO][acceleratesupport.py:12]:OpenGL_accelerate module loaded
[WARNING][arraydatatype.py:17]:Unable to load ArrayDatatype accelerator from OpenGL_accelerate
[WARNING][converters.py:122]:Unable to load converters accelerators (wrapper, arraydatatype) from OpenGL_accelerate
[WARNING][arrayhelpers.py:22]:Unable to load arrayhelpers accelerator from OpenGL_accelerate
[WARNING][numpymodule.py:28]:Unable to load numpy_formathandler accelerator from OpenGL_accelerate
[INFO][formathandler.py:54]:Unable to load registered array format handler numeric:
Traceback (most recent call last):
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/arrays/formathandler.py", line 44, in loadPlugin  
    plugin_class = entrypoint.load()
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/plugins.py", line 14, in load (self is a FormatHandler)
    return importByName( self.import_path )
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/plugins.py", line 28, in importByName  
    module = __import__( ".".join(moduleName), {}, {}, moduleName)
  File "/home/seann/.local/lib/python2.7/site-packages/PyOpenGL-3.0.1-py2.7.egg/OpenGL/arrays/numeric.py", line 15, in <module>  
    raise ImportError( """No Numeric module present: %s"""%(err))
ImportError: No Numeric module present: No module named Numeric

[INFO][settings.py:100]:Loading app settings from /home/seann/mcedit_v2/MCEdit User Data/mcedit2.ini
[INFO][loadablechunks.py:17]:Making checkerboard texture...
[INFO][minecraftinstall.py:235]:Checking install at /home/seann/.minecraft
[INFO][minecraftinstall.py:242]:Found versions:
[u'1.6.4', u'1.7.4', u'1.8.8', u'1.8', u'1.7.2', u'1.7.10', u'1.8.7', u'1.8.1', u'1.7.9']
[INFO][minecraftinstall.py:235]:Checking install at /home/seann/.minecraft
[INFO][minecraftinstall.py:242]:Found versions:
[u'1.6.4', u'1.7.4', u'1.8.8', u'1.8', u'1.7.2', u'1.7.10', u'1.8.7', u'1.8.1', u'1.7.9']
[WARNING][editorapp.py:92]:UserFilesDirectory: /home/seann/mcedit_v2/MCEdit User Data
[INFO][editorapp.py:100]:Loaded translator.
[INFO][editorapp.py:109]:Loaded stylesheet.
[INFO][editorapp.py:121]:Loaded main window.
[INFO][editorapp.py:177]:Loaded panels.
[INFO][editorapp.py:220]:Loaded debug widgets.
[INFO][editorapp.py:270]:Loaded menus.
[INFO][worldlist.py:94]:Size hint: PySide.QtCore.QSize(145, 79)
[INFO][worldlist.py:95]:Size : PySide.QtCore.QSize(145, 100)
[INFO][worldlist.py:292]:Scanning /home/seann/.minecraft/saves for worlds...
[INFO][minecraftinstall.py:287]:Created ResourceLoader with search path:
/home/seann/.minecraft/versions/1.8.8/1.8.8.jar (6585891791e0d75dc54cfb547a866db0)
[INFO][qglcontext.py:52]:Obtained a GL context with this format:
Valid: True
Version: 1.3
Hardware Accelerated: True
Depth buffer: True, 24
Double buffer: True

Driver info:
GL_VERSION: 3.0 Mesa 11.0.4 (3, 0)
GL_VENDOR: 'X.Org'
GL_RENDERER: 'Gallium 0.4 on AMD TURKS (DRM 2.40.0, LLVM 3.6.0)'

[INFO][worldview.py:873]:LAYER Blocks VISIBLE True
[INFO][worldview.py:873]:LAYER MonsterLocations VISIBLE False
[INFO][worldview.py:873]:LAYER Items VISIBLE False
[INFO][worldview.py:873]:LAYER TileEntities VISIBLE False
[INFO][worldview.py:873]:LAYER TileEntityLocations VISIBLE False
[INFO][worldview.py:873]:LAYER CommandBlockColors VISIBLE True
[INFO][worldview.py:873]:LAYER CommandBlockLocations VISIBLE False
[INFO][worldview.py:873]:LAYER ItemFrames VISIBLE False
[INFO][worldview.py:873]:LAYER TileTicks VISIBLE False
[INFO][worldview.py:873]:LAYER ChunkSections VISIBLE False
[INFO][worldview.py:873]:LAYER HeightMap VISIBLE False
[INFO][worldview.py:213]:Changing MinimapWorldView(u'None') to dimension WorldEditorDimension(dimName='', adapter=AnvilWorldAdapter(u'/home/seann/.minecraft/saves/Link to -12_2'))
[INFO][chunkloader.py:155]:Added: client MinimapWorldView(u'Link to -12_2: ')
[INFO][worldlist.py:354]:Centering on spawn position.
[INFO][worldlist.py:369]:Switched world view
[INFO][editorapp.py:280]:Loaded world list.
[INFO][editorapp.py:299]:Loaded status bar.
[INFO][editorapp.py:594]:Updating recent files menu: (1) [u'/home/seann/.minecraft/saves/Seann']
[INFO][editorapp.py:609]:Adding Seann
[INFO][editorapp.py:306]:Loaded settings.
[WARNING][configureblocksdialog.py:119]:Failed to read definitions file defined_blocks.json
[INFO][editorapp.py:328]:Loaded app dialogs.
[INFO][editorapp.py:336]:Loading timer started
[INFO][editorapp.py:346]:Finding plugins
[INFO][plugins.py:204]:Loading plugins from /home/seann/mcedit_v2/src/plugins
[INFO][plugins.py:111]:Trying to load plugin from koch.py
[INFO][plugins.py:122]:Loaded koch.py (Koch Snowflake)
[INFO][plugins.py:111]:Trying to load plugin from hilbert.py
[INFO][plugins.py:122]:Loaded hilbert.py (Hilbert Curve)
[INFO][plugins.py:111]:Trying to load plugin from storagedrawers.py
[INFO][plugins.py:122]:Loaded storagedrawers.py (storagedrawers)
[INFO][plugins.py:111]:Trying to load plugin from city.py
[INFO][plugins.py:122]:Loaded city.py (GFX: City)
[INFO][editorapp.py:363]:Opening worlds from command line.
[INFO][worldlist.py:382]:Removing view from WorldListWidget
[INFO][minecraftinstall.py:287]:Created ResourceLoader with search path:
/home/seann/.minecraft/versions/1.8.8/1.8.8.jar (6585891791e0d75dc54cfb547a866db0)
[INFO][qglcontext.py:52]:Obtained a GL context with this format:
Valid: True
Version: 1.3
Hardware Accelerated: True
Depth buffer: True, 24
Double buffer: True

Driver info:
GL_VERSION: 3.0 Mesa 11.0.4 (3, 0)
GL_VENDOR: 'X.Org'
GL_RENDERER: 'Gallium 0.4 on AMD TURKS (DRM 2.40.0, LLVM 3.6.0)'

[INFO][worldview.py:873]:LAYER Blocks VISIBLE True
[INFO][worldview.py:873]:LAYER MonsterLocations VISIBLE False
[INFO][worldview.py:873]:LAYER Items VISIBLE False
[INFO][worldview.py:873]:LAYER TileEntities VISIBLE False
[INFO][worldview.py:873]:LAYER TileEntityLocations VISIBLE False
[INFO][worldview.py:873]:LAYER CommandBlockColors VISIBLE True
[INFO][worldview.py:873]:LAYER CommandBlockLocations VISIBLE False
[INFO][worldview.py:873]:LAYER ItemFrames VISIBLE False
[INFO][worldview.py:873]:LAYER TileTicks VISIBLE False
[INFO][worldview.py:873]:LAYER ChunkSections VISIBLE False
[INFO][worldview.py:873]:LAYER HeightMap VISIBLE False
[INFO][worldview.py:213]:Changing MinimapWorldView(u'None') to dimension WorldEditorDimension(dimName='', adapter=AnvilWorldAdapter(u'/home/seann/.minecraft/saves/Link to -12_2'))
[INFO][chunkloader.py:155]:Added: client MinimapWorldView(u'Link to -12_2: ')
[INFO][worldlist.py:354]:Centering on spawn position.
[INFO][worldlist.py:369]:Switched world view
[INFO][worldeditor.py:420]:Scanning for regions in /home/seann/.minecraft/saves/Link to -12_2...
[INFO][worldeditor.py:429]:Dim : Found 4096 chunks in 0.10 seconds.
[INFO][extensions.py:23]:OpenGL Version: 3.0 Mesa 11.0.4
```

**Comment by codewarrior0** [comment]
Ugh, I've got no idea about that. Ever used `git bisect` before? Figure out what was the last revision that worked for you, and then:

```
git bisect start gd4f1656 <last working revision>
git bisect run mcedit2 -eval 'raise SystemExit'
```

**Comment by EvilSupahFly** [comment]
No idea what the last working version was. I should keep a backup somewhere when I get a version working. So, I've moved it to a new folder, and I'm starting over, with a fresh copy and reinstalling from scratch - including the required python libraries and other dependencies using `pip install -r requirements.txt` to make sure it's not something on my end. I'll try again after that's done and see what happens.

**Comment by codewarrior0** [comment]
Never mind. I can reproduce this. The problem started with 81c65fb925e3133332869a74e1265cac0a9cae49.

I knew it was too good to be true!

**Comment by codewarrior0** [comment]
Should be fixed in 39a717b3cab5dd8366ed8542a070e4120386eb92

**Comment by EvilSupahFly** [comment]
Nice work. Loads again! Thanks!
